libs/designsystem/src/lib/components/checkbox/checkbox.component.spec.ts

```diff
@@ -47,20 +59,50 @@ describe('CheckboxComponent', () => {
       expect(spectator.element).toHaveClass('attention-level1');
     });
 
+    it('should have correct icon styling when attentionLevel is 1', () => {
+      spectator.setInput('attentionLevel', '1');
+      spectator.detectChanges();
+      expect(ionCheckbox).toHaveComputedStyle({
+        '--checkmark-color': getColor('black'),
+        '--background-checked': getColor('success'),
+        '--border-color-checked': getColor('success'),
+      });
+    });
+
     it('should set the [checked] input on ion-checkbox', () => {
       spectator.setInput('checked', true);
-      expect(checkbox.checked).toBe(true);
+      expect(ionCheckbox.checked).toBe(true);
 
       spectator.setInput('checked', false);
-      expect(checkbox.checked).toBe(false);
+      expect(ionCheckbox.checked).toBe(false);
     });
 
     it('should set the [disabled] input on ion-checkbox', () => {
       spectator.setInput('disabled', true);
-      expect(checkbox.disabled).toBe(true);
+      expect(ionCheckbox.disabled).toBe(true);
 
       spectator.setInput('disabled', false);
-      expect(checkbox.disabled).toBe(false);
+      expect(ionCheckbox.disabled).toBe(false);
+    });
+
+    it('should have correct text styling when disabled', () => {
+      spectator.setInput('disabled', true);
+      spectator.detectChanges();
+      expect(spectator.element).toHaveComputedStyle({
+        color: getTextColor('semi-dark'),
+      });
+    });
+
+    it('should have correct icon styling when disabled', () => {
+      spectator.setInput('disabled', true);
+      spectator.detectChanges();
+      expect(ionCheckbox).toHaveComputedStyle({
+        '--checkmark-color': getColor('semi-dark'),
+        '--background': getColor('semi-light'),
+        '--background-checked': getColor('semi-light'),
+        '--border-color': getColor('medium'),
+        '--border-color-checked': getColor('semi-light'),
+      });
     });
 
     it('should only set error class when [hasError] input is true', () => {
```

Comment:
Consider if this class check is still relevant after adding the `'should have correct icon styling when checkbox has error'` test...